src/endpoint.rs

```diff
@@ -139,4 +148,43 @@ impl<S: MutEventSubscriber> RemoteEndpoint<S> {
         // own a `b: Box<T>` and call `*b`, the box goes away and we get the `T` inside.
         *result_box
     }
+
+    /// Call the specified closure on the associated local/remote `EventManager` (provided as a
+    /// `SubscriberOps` trait object), and discard the result. This method only fires
+    /// the request but do not wait for result, so it may be called from the same thread where
+    /// the event loop runs.
+    pub fn fire<F>(&self, f: F) -> Result<()>
+    where
+        F: FnOnce(&mut dyn SubscriberOps<Subscriber = S>) -> ErasedResult + Send + 'static,
+    {
+        // We erase the return type of `f` by moving and calling it inside another closure which
+        // hides the result as an `ErasedResult`. This allows using the same channel send closures
+        // with different signatures (and thus different types) to the remote `EventManager`.
+        let fnbox = Box::new(
+            move |ops: &mut dyn SubscriberOps<Subscriber = S>| -> ErasedResult { Box::new(f(ops)) },
+        );
+
+        // Send the message requesting the closure invocation.
+        self.send(FnMsg {
+            fnbox,
+            sender: None,
+        })
+    }
+}
+
+/// Enables interactions with an `EventManager` that runs on a different thread of execution.
+#[derive(Clone)]
+pub struct RemoteKicker {
+    // Used to notify the `EventManager` to wakeup.
+    event_fd: Arc<EventFd>,
+}
+
+impl RemoteKicker {
+    /// Kick the worker thread to wake up from the epoll event loop.
+    pub fn kick(&self) -> Result<()> {
```

Comment:
This method is quite interesting in the context of the event manager. Can you share some more details regarding what do you envision at its primary uses?

Also, just wondering, is it different from calling `fire` with a closure that does nothing? Generally speaking, is the functionality provided by a `RemoteKicker` required given the methods that are exposed by a `RemoteEndpoint`? If the convenience of having a simple `kick`method with no parameters is important, can we just move this to `RemoteEndpoint` as well, and have it maybe invoke `fire` as part of the implementation?

Reply:
Sure, I will merge the RemoteKicker into RemoteEndpoint, it simplifies the interfaces.
The scenario is simple, a vCPU thread wants to wake up the IO thread from sleep, so the IO thread will release and reacquire some locks. Otherwise the vCPU thread will block waiting for the locks.